consensus/src/block_graph.rs

```diff
@@ -3107,8 +3119,9 @@ impl BlockGraph {
         inherited_incomp_count: usize,
         block_ledger_changes: LedgerChanges,
         operation_set: OperationHashMap<(usize, u64)>,
-        endorsement_ids: Vec<EndorsementId>,
+        endorsement_ids: EndorsementHashMap<u32>,
         addresses_to_operations: AddressHashMap<OperationHashSet>,
+        addresses_to_endorsements: AddressHashMap<EndorsementHashSet>,
         roll_updates: RollUpdates,
         production_events: Vec<(u64, Address, bool)>,
     ) -> Result<(), ConsensusError> {
```

Comment:
Too much arguments in this function. We can use right now a appropriate structure

Reply:
https://github.com/massalabs/massa/issues/1232 😅